src/super_gradients/module_interfaces/exportable_detector.py

```diff
@@ -0,0 +1,399 @@
+            preprocessing = nn.Sequential(CastTensorTo(model_type), preprocessing)
+            input_image_dtype = input_image_dtype or torch.uint8
+        else:
+            preprocessing = None
+            input_image_dtype = input_image_dtype or model_type
+
+        # This variable holds the output names of the model.
+        # If postprocessing is enabled, it will be set to the output names of the postprocessing module.
+        output_names: Optional[List[str]] = None
+
+        if isinstance(postprocessing, nn.Module):
+            # If a user-specified postprocessing module is provided, we will attach is to the model and not
+            # attempt to attach NMS step, since we do not know what the user-specified postprocessing module does,
+            # and what outputs it produces.
+            attach_nms_postprocessing = False
+        elif postprocessing is True:
+            attach_nms_postprocessing = True
+            postprocessing_kwargs = postprocessing_kwargs or {}
+            postprocessing_kwargs["num_pre_nms_predictions"] = num_pre_nms_predictions
+            postprocessing: AbstractObjectDetectionDecodingModule = model.get_decoding_module(**postprocessing_kwargs)
+
+            output_names = postprocessing.get_output_names()
+            num_pre_nms_predictions = postprocessing.num_pre_nms_predictions
+            max_predictions_per_image = max_predictions_per_image or num_pre_nms_predictions
+
+            nms_threshold = nms_threshold or getattr(model, "_default_nms_iou", None)
+            if nms_threshold is None:
+                raise ValueError(
+                    "nms_threshold is not specified and cannot be inferred from the model. "
+                    "Please specify the nms_threshold explicitly: model.export(..., nms_threshold=0.5)"
+                )
+
+            confidence_threshold = confidence_threshold or getattr(model, "_default_nms_conf", None)
+            if confidence_threshold is None:
+                raise ValueError(
+                    "confidence_threshold is not specified and cannot be inferred from the model. "
+                    "Please specify the confidence_threshold explicitly: model.export(..., confidence_threshold=0.5)"
+                )
+
+            if max_predictions_per_image > num_pre_nms_predictions:
+                raise ValueError(
+                    f"max_predictions_per_image={max_predictions_per_image} is greater than "
+                    f"num_pre_nms_predictions={num_pre_nms_predictions}. "
+                    f"Please specify max_predictions_per_image <= {num_pre_nms_predictions}."
+                )
+        else:
+            attach_nms_postprocessing = False
+            postprocessing = None
+
+        if hasattr(model, "prep_model_for_conversion"):
+            model.prep_model_for_conversion(**prep_model_for_conversion_kwargs)
+
+        if quantization_mode == ExportQuantizationMode.INT8:
```

Comment:
Let's check if the model has already quantised layers inside it (i.e QuantConv2d etc), and if so skip this part.
Then we can call export right after Trainer's quantization methods.

Reply:
Fixed in 57890f38b159f9fae4d745c5cf97b90937d16311